packages/babel-core/test/config-loading.js

```diff
@@ -318,13 +321,12 @@ describe("@babel/core config loading", () => {
         plugins: opts.plugins.map(([plg, opt]) => [plg, { ...opt }]),
       }).options;
       expect(options2.plugins.length).toBe(options1.plugins.length);
-
+      expect(options2.plugins[2]).not.toBe(options1.plugins[2]);
       for (let i = 0; i < options2.plugins.length; i++) {
         if (i === 2) {
-          expect(options2.plugins[i]).not.toBe(options1.plugins[i]);
-        } else {
-          expect(options2.plugins[i]).toBe(options1.plugins[i]);
+          continue;
```

Comment:
This is sort-of cheating. The `expect` is still implicitly guarded. However I can't figure out a better way except constructing a list of valid `i`, which seems to be overkills here.

/cc @SimenB

Reply:
Personal preference is mostly ignore patterns even if I generally agree with the lint rule.

@g-rath thoughts on this one?

Reply:
I think I like how it looks here - but that could also be because I'm bias against `else` 😅 

Another alternative would be to do the comparison yourself and compare to `true`:

```
const plugin1 = options1.plugins[i];
const plugin2 = options2.plugins[i];

expect(
  i === 2 
    ? plugin2 !== plugin1
    : plugin2 === plugin1
).toBe(true);
```

I'm pretty sure we don't have any rules that'll complain on that (tho maybe we should?) - but one of the big downsides of course is that `jest` won't be as helpful with it's error messages if the test failed, since it'll always just complain "expected value to be true"